projects/plugins/jetpack/modules/masterbar/wp-posts-list/class-posts-list-page-notification.php

```diff
@@ -0,0 +1,129 @@
+	 */
+	private $posts_page_id;
+
+	/**
+	 * If the Post_list contains the site's Posts Page
+	 *
+	 * @var bool
+	 */
+	private $is_page_in_list = false;
+
+	/**
+	 * Class instance.
+	 *
+	 * @var Posts_List_Page_Notification|null
+	 */
+	private static $instance = null;
+
+	/**
+	 * Posts_List_Page_Notification constructor.
+	 *
+	 * @param string $posts_page_id The Posts page configured in WordPress.
+	 */
+	public function __construct( $posts_page_id ) {
+		add_action( 'init', array( $this, 'init_actions' ) );
+
+		$this->posts_page_id = '' === $posts_page_id ? null : (int) $posts_page_id;
+	}
+
+	/**
+	 * Add in all hooks.
+	 */
+	public function init_actions() {
+		\add_filter( 'map_meta_cap', array( $this, 'disable_posts_page' ), 10, 4 );
+		\add_filter( 'post_class', array( $this, 'add_posts_page_css_class' ), 10, 3 );
+		\add_action( 'admin_print_footer_scripts-edit.php', array( $this, 'add_notification_icon' ) );
+	}
+
+	/**
+	 * Creates instance.
+	 *
+	 * @return Posts_List_Page_Notification
+	 */
+	public static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self( \get_option( 'page_for_posts' ) );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Disable editing and deleting for the page that is configured as a Posts Page.
+	 *
+	 * @param array  $caps Array of capabilities.
+	 * @param string $cap The current capability.
+	 * @param string $user_id The user id.
+	 * @param array  $args Argument array.
+	 * @return array
+	 */
+	public function disable_posts_page( $caps, $cap, $user_id, $args ) {
+		if ( 'edit_post' !== $cap && 'delete_post' !== $cap ) {
+			return $caps;
+		}
+
+		if ( isset( $args[0] ) && $this->posts_page_id === $args[0] ) {
+			$caps[] = 'do_not_allow';
+		}
+
+		return $caps;
+	}
+
+	/**
+	 * Adds a CSS class on the page configured as a Posts Page.
+	 *
+	 * @param array  $classes A list of CSS classes.
+	 * @param string $class A CSS class.
+	 * @param string $post_id The current post id.
+	 * @return array
+	 */
+	public function add_posts_page_css_class( $classes, $class, $post_id ) {
+		if ( $this->posts_page_id !== $post_id ) {
+			return $classes;
+		}
+
+		$this->is_page_in_list = true;
+
+		$classes[] = 'posts-page';
+
+		return $classes;
+	}
+
+	/**
+	 * Add a info icon on the Posts Page letting the user know why they cannot delete and remove the page.
+	 */
+	public function add_notification_icon() {
+		// No need to add the JS since the site is not configured with a Posts Page or the current listview doesn't contain the page.
+		if ( null === $this->posts_page_id || ! $this->is_page_in_list ) {
+			return;
+		}
+
+		$text_notice = __( 'The content of your latest posts page is automatically generated and cannot be edited', 'jetpack' );
+		?>
+		<script>
+			document.querySelector(".posts-page .check-column").innerHTML = '<span title="<?php echo esc_html( $text_notice ); ?>" style="margin-left: 8px" class="dashicons dashicons-info-outline"></span>';
```

Comment:
Is the `title` attribute visible on mobile devices? I tried "tapping" the info icon, but the message doesn't show up so it seems we may need an alternative for mobiles?

<img width="438" alt="Screen Shot 2021-05-21 at 14 55 23" src="https://user-images.githubusercontent.com/1233880/119141404-81e96b00-ba45-11eb-85d0-d5c41730ee97.png">

Additionally, I'd suggest to make the icon a bit wider on mobile viewports, so it's aligned with the checkboxes.

Reply:
Yeah, just noticed as well. The problem on mobile with the  title hover is a bit trickier, WPCore doesn't have a tooltip library I could use :(

Reply:
> we may need an alternative for mobiles?

I'm happy to leave that for a follow-up though

Reply:
> WPCore doesn't have a tooltip library I could use :(

Maybe jQuery UI? Tooltips? Seems to be available by default on WordPress: https://developer.wordpress.org/reference/functions/wp_enqueue_script/#default-scripts-and-js-libraries-included-and-registered-by-wordpress

Reply:
Lovely! The new tooltip using purse HTML/JS/CSS looks great on both desktop and mobile viewports.

Desktop | Viewports
--- | ---
<img width="335" alt="Screen Shot 2021-05-21 at 17 06 16" src="https://user-images.githubusercontent.com/1233880/119158949-0db7c300-ba57-11eb-9bcf-b9748aeb1cae.png"> | <img width="416" alt="Screen Shot 2021-05-21 at 17 06 26" src="https://user-images.githubusercontent.com/1233880/119158986-18725800-ba57-11eb-8e50-d6640ce7d93f.png">